### PR TITLE
Add PypeIt to conda-forge

### PR DESCRIPTION
**Pull request by finagle29**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/pypeit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  host:
+    - python
+    - pip
+    - extension-helpers >=0.1
+  run:
+    - python
+    - numpy >=1.16.0
+    - astropy >=4.0
+    - scipy >=1.4
+    - matplotlib >=3.1
```
Comment (scopatz) [inline recipes/pypeit/meta.yaml:29]: This should be matplotlib-base

File: recipes/pypeit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    - pyqt >=5
+    - pyyaml >=5.1
+    - configobj >=5.0.6
+    - scikit-learn >=0.20
+    - ipython >=7.2.0
+    - ginga >=3.0
+    - packaging >=19.0
+    - linetools >=0.2
+    - extension-helpers >=0.1
+    - pytest >=3.0.7
```
Comment (scopatz) [inline recipes/pypeit/meta.yaml:40]: Is pytest really a runtime dep?

File: recipes/pypeit/meta.yaml (hunk 1 above)
Comment (finagle29) [reply recipes/pypeit/meta.yaml:29]: Thanks! Will fix in next commit.

File: recipes/pypeit/meta.yaml (hunk 2 above)
Comment (finagle29) [reply recipes/pypeit/meta.yaml:40]: Currently, it is.

File: recipes/pypeit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - extension-helpers >=0.1
+  run:
+    - {{ cdt('mesa-libgl') }}  # [linux]
```
Comment (scopatz) [inline recipes/pypeit/meta.yaml:25]: This should probably be in build

Comment (finagle29) [reply recipes/pypeit/meta.yaml:25]: Actually, I'm not sure if it's necessary at all. I had added it to try and fix libgl.so.1 not being found during the test phase, but I think that that might actually be fixed by ``yum_requirements.txt``. I can try removing this, though.

**Pull request merged**